### PR TITLE
adds validation to the response body from pusher to ensure that error…

### PR DESCRIPTION
**Pull request by jonas-zipprick**
…s are handled gracefully. regarding #7

### CODE REVIEW

File: lib/pusher/http_client.ex [hunk 1]
```diff
@@ -11,7 +11,8 @@ defmodule Pusher.HttpClient do
   end
 
   defp process_response_body(body) do
-    unless body == "", do: body |> JSX.decode!, else: nil
+    :ok = validate_response_body(body)
```
Comment (edgurgel) [inline lib/pusher/http_client.ex]: What if we do 
```elixir
case JSX.decode do
  {:ok, json} -> ...
  {:error, _ } -> {:invalid_response, ...}
end
```

Comment (jonas-zipprick) [reply lib/pusher/http_client.ex]: The actual reason for the error is contained in the response body (e.g. Invalid auth key).
If we do it the way you propose then it wont be part of the error message.
Thats also why i am returning the body as part of the error message in line 36.

Comment (edgurgel) [reply lib/pusher/http_client.ex]: That's fine I'm just saying that using `JSX.decode` will say if the body was JSON or not. If you get :error then you know that the JSON decoding didn't work

**Comment by edgurgel** [comment]
Fixed by #13

**Pull request closed without merging**